Server/VikingVault.Services/AdminService.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Text;
+using VikingVault.DataAccess;
+using VikingVault.Services.Abstractions;
+
+namespace VikingVault.Services
+{
+    public class AdminService : IAdminService
+    {
+        private readonly VikingVaultDbContext _dbContext;
+
+        public AdminService(VikingVaultDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public bool IsAdmin(string token)
+        {
+            var tokenObject = new JwtSecurityToken(token);
+            string userId = tokenObject.Payload["Id"].ToString();
+            var returnedUser = _dbContext.User.SingleOrDefault(u => u.Id == Int32.Parse(userId));
```

Comment:
Just a warning here, when you use the `.Parse()`, it can fail to parse, so keep that in mind. It's usually better to use `.TryParse()` as the default parsing method. You don't need to change anything here, just keep that in mind (you probably knew that already but reminders are good 😅 )